hanjaelee/app.js

```diff
@@ -52,20 +53,172 @@ app.post('/signup', (req, res) => {
       profile_image,
       password
     ) VALUES (?, ?, ?, ?);`
-  const rawData = database.query(rawQuery, [name, email, profileImage, password])
 
-  rawData
-    .then((data) => {
-      if (data.affectedRows) {
-        res.status(201).json({ message: 'userCreated' })
-        return
-      }
+  try {
+    const rawData = await database.query(rawQuery, [name, email, profileImage, password])
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'userCreated' })
+    }
+    return res.status(202).json({ message: 'userNotCreated' })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 3 게시글 등록하기
+app.post('/post', async (req, res) => {
+  const { title, content, userId } = req.body
+
+  const rawQuery =`
+    INSERT INTO posts (
+      title,
+      content,
+      user_id
+    ) VALUES (?, ?, ?);`
 
-      res.status(400).json({ message: 'userIsNotCreated' })
+  try {
+    const rawData = await database.query(rawQuery, [title, content, userId])
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'postCreated' })
+    }
+    return res.status(202).json({ message: 'postnotcreated' })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 4 모든 게시물 가져오기
+app.get('/posts', async (req, res) => {
+  const rawQuery = `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts AS p
+    INNER JOIN users AS u
+    ON p.user_id = u.id
+    `
+  try {
+    await database.query(rawQuery, (err, rows) => {
+      const result = {}
+      const data = rows
+      result['data'] = data
+      res.status(200).json(result)
     })
-    .catch((error) => {
-      res.status(400).json({ message: error.sqlMessage })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 5 유저 한명이 작성한 모든 게시물 가져오기
+app.get('/posts/:userId', async (req, res) => {
+  const { userId } = req.params
+  const rawQuery = `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      JSON_ARRAYAGG(JSON_OBJECT(
+                    'postingId', p.id,
+                    'postingImageUrl', p.image_url,
+                    'postingContent', p.content)) AS posting
+    FROM posts AS p
+    INNER JOIN users AS u
+    ON p.user_id = u.id WHERE u.id = ${userId}
+    GROUP BY u.id;`
```

Comment:
- 해당 부분도 `try { ... }` 구문 안에 넣어도 될 것 같아요.
-`rawQuery`에 대한 포매팅도 수정하면 가독성이 더 좋아질 것 같습니다.
- `?` 매핑을 사용할 수 있게 수정해주세요!

```suggestion
  const { userId } = req.params
  const rawQuery = `
    SELECT
      u.id AS userId,
      u.profile_image AS userProfileImage,
      JSON_ARRAYAGG(
        JSON_OBJECT(
          'postingId', p.id,
          'postingImageUrl', p.image_url,
          'postingContent', p.content
          )
      ) AS postings
    FROM posts AS p
    INNER JOIN users AS u
    ON p.user_id = u.id WHERE u.id = ?
    GROUP BY u.id;`
```